R/travis.R

```diff
@@ -42,6 +42,12 @@ TravisCI <- R6Class(
         list(crayon.enabled = TRUE),
         cat_line(code)
       )
+    },
+    is_travis = function() {
+     self$is_env("TRAVIS", true)
```

Comment:
This should return a constant `TRUE`, and the implementation in the base class should return a constant `FALSE`. Similarly for `is_appveyor()`.

Only one instance of `CI` is created, and it's already conditioned on the env variables set.

Reply:
ok, so going back to `inherits(ci(), <CI>)`? This makes use of the information already stored in the `ci()` instance

Reply:
This method should look like this:

```
is_travis = function() {
  TRUE
}
```

But only for the `TravisCI` class -- the other classes use the implementation in the base class, which returns `FALSE` .

Reply:
Hm, but the `inherits(ci())` way would still do its job here, right? Sure, your proposed way is cleaner and is only needed once per CI..

Hmz, those little helpers are really confusing me sometimes.

Reply:
Yes, it would work, but look a bit like checking `class(structure(x, class = "TravisCI"))` . This particular method already knows that the object for which it has been called is of class `"TravisCI"` -- this method isn't called for objects of other classes.

We defined `ci_xxx()` as a shortcut for `ci()$xxx()` .